homeassistant/components/sensor/serial.py

```diff
@@ -71,7 +80,18 @@ def serial_read(self, device, rate, **kwargs):
             url=device, baudrate=rate, **kwargs)
         while True:
             line = yield from reader.readline()
-            self._state = line.decode('utf-8').strip()
+            line = line.decode('utf-8').strip()
+
+            try:
+                self._attributes = json.loads(line)
+            except json.JSONDecodeError:
+                self._attributes = []
```

Comment:
This doesn't make much sense. There is no reason to have the state duplicated into an attribute if only one value, the state, is available. Beside that will this lead to a `TypeError` for the attributes. 

Something like this could work if only one value is received:

```python
            try:
                data = json.loads(line)
            except json.JSONDecodeError:
                _LOGGER.error("Invalid JSON data received: %s", data)

            if isinstance(data, dict):
                self._attributes = data
```

Reply:
the except makes no sense, since when you only have a string it will necessary log an error (which is not a real error).

so now I just pass the exception altogether ignoring when it's not a JSON

Reply:
Well, a string is not valid JSON.

Being for tolerant for data that is coming from a device is always good.